### PR TITLE
feat: add method RequestBuilder.resolveRequestUrl

### PR DESCRIPTION
**Pull request by padamstx**
This commit adds method resolveRequestURL to the RequestBuilder class
as a replacement for the constructHttpUrl() method.

### CODE REVIEW

**Comment by padamstx** [comment]
Removed reviewers because I noticed a couple things that need to be fixed.

File: pom.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@
         <compiler-plugin-version>3.8.0</compiler-plugin-version>
         <okhttp3-version>3.14.9</okhttp3-version>
         <logback-version>1.2.3</logback-version>
-        <guava-version>27.1-android</guava-version>
```
Comment (jorge-ibm) [inline pom.xml:45]: 🤔

**Review by jorge-ibm — APPROVED** [review]
LGTM

File: pom.xml (hunk 1 above)
Comment (padamstx) [reply pom.xml:45]: while the old version # refers to android, that version supported java 7+, but we no longer need java 7 support so I just moved us up to the latest.

**Pull request merged**